AzureFunction.cs

```diff
@@ -0,0 +1,404 @@
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using Wishlist;
+using Azure.Data.Tables;
+using System.Collections.Generic;
+using Wishlist.Models;
+using System.Linq;
+using Wishlist.Models.AppModels;
+using Wishlist.Models.Entities;
+using System.Linq.Expressions;
+using Wishlist.Models.Dtos;
+
+namespace Wishlist
+{
+    public class AzureFunction
+    {
+       private  readonly BlobStorageContext blobStorage;
+       private readonly TableStorageContext tableStorage;
+
+      public AzureFunction(BlobStorageContext _blobStorage,TableStorageContext _tableStorage)
+      {
+          blobStorage=_blobStorage;
+          tableStorage = _tableStorage;
+      }
+
+
+        [FunctionName("UploadFile")]
+        public async Task<IActionResult> UploadFile(
+            [HttpTrigger(AuthorizationLevel.Function, "post", Route = null)] HttpRequest req,
+            ILogger log)
+        {
+            string id = req.Query["productId"];
+            var file = req.Form.Files[0];
+            
+            string blobName = await GetBlobNameAsync(id);
+            if(blobName is null)
+            {
+                return new NotFoundResult();
+            }
+
+            using(var stream = file.OpenReadStream()){
+                await blobStorage.UploadStreamAsync(WC.ProdutBlog,blobName,stream);
+            }
+            return new OkResult();
+        }
+
+
+        private async Task<string> GetBlobNameAsync(string id)
+        {
+            ProductEntity product = await tableStorage.GetEntityByKeyAsync<ProductEntity>(WC.ProdcutTable,id);
+            if(product is null)
+            {
+                return null;
+            }
+            string blobName = $"{product.PartitionKey}-{product.RowKey}";
+            return blobName;
+        }

```

Comment:
In this file we try to only define our http endpoints and add helper function like this to a different class. I would recommend adding this to your `BlobStorageContext` class.